frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -850,11 +857,18 @@ export function stream(url: string, cb: StreamResultsCb, args: StreamArgs) {
   function cancel() {
     if (connection) connection.close();
     isCancelled = true;
+    OPEN_SOCKET_CONNECTIONS.delete(url);
```

Comment:
This means that if we have currently 2 users (places in the code, not actual different people) for this connection, and one cancels the connection, the others will have it cancelled too.

See my other comment for keeping track of the list of callbacks/onFails, and this list can act as a ref count for making sure we only effectively stop the connection when there's no list of tracked callers anymore.